packages/compass-e2e-tests/tests/atlas-login.test.ts

```diff
@@ -0,0 +1,425 @@
+import type { CompassBrowser } from '../helpers/compass-browser';
+import {
+  beforeTests,
+  afterTests,
+  afterTest,
+  Selectors,
+} from '../helpers/compass';
+import type { Compass } from '../helpers/compass';
+import type { OIDCMockProviderConfig } from '@mongodb-js/oidc-mock-provider';
+import { OIDCMockProvider } from '@mongodb-js/oidc-mock-provider';
+import path from 'path';
+import { expect } from 'chai';
+import { promisify } from 'util';
+import { createNumbersCollection } from '../helpers/insert-data';
+
+const DEFAULT_TOKEN_PAYLOAD = {
+  expires_in: 3600,
+  payload: {
+    groups: ['testgroup'],
+    sub: 'testuser',
+    aud: 'resource-server-audience-value',
+  },
+};
+
+function getTestBrowserShellCommand() {
+  return `${process.execPath} ${path.resolve(
+    __dirname,
+    '..',
+    'fixtures',
+    'curl.js'
+  )}`;
+}
+
+describe('Atlas Login', function () {
+  let compass: Compass;
+  let browser: CompassBrowser;
+  let oidcMockProvider: OIDCMockProvider;
+  let originalDisableKeychainUsage: string | undefined;
+  let getTokenPayload: OIDCMockProviderConfig['getTokenPayload'];
+
+  before(async function () {
+    originalDisableKeychainUsage =
+      process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE;
+
+    function isAuthorised(req: { headers: { authorization?: string } }) {
+      const [, token] = req.headers.authorization?.split(' ') ?? [];
+      // We can't check that the issued token is the one received by oidc-plugin
+      // so we are only checking that it was passed and assuming that it's good
+      // enough of a validation
+      return !!token;
+    }
+
+    oidcMockProvider = await OIDCMockProvider.create({
+      getTokenPayload(metadata) {
+        return getTokenPayload(metadata);
+      },
+      async overrideRequestHandler(_url, req, res) {
+        const url = new URL(_url);
+        const end = promisify(res.end.bind(res));
```

Comment:
Not a big deal, but it would also be just fine to do `res.end()` directly, shouldn’t make any difference here and might look a bit more idiomatic I guess.

Reply:
Yeah, it's the difference between `writableEnded` and `writableFinished` ... naming is hard 😅

Reply:
Removed async in https://github.com/mongodb-js/compass/pull/4900/commits/c89a4e09004f9437441e74f7775c3619ef17076a